src/compute_orientorder_atom.cpp

```diff
@@ -621,68 +637,92 @@ double ComputeOrientOrderAtom::associated_legendre(int l, int m, double x)
+
+        // Determine number of elements in symmetry group of (m1,m2,m3)
+        // Concise determination exploiting (m1,m2,m3) loop structure.
+        int pfac;
+        if (m1 == 0) pfac = 1; // m1 = m2 = m3 = 0
+        else if (m2 == 0 || m2 == m3) {
+          // reduced group when only 3 permutations, or sign inversion
+          // is equivalent to permutation
+          pfac = 6;
+        } else pfac = 12; // 6 permutations * 2 signs
+
+        w3jlist[widx_count] = w3j(l,m1,m2,m3) * pfac;
+        widx_count++;
+      }
+    }
   }
-  idxcg_max = idxcg_count;
-  memory->destroy(cglist);
-  memory->create(cglist, idxcg_max, "computeorientorderatom:cglist");
+}
 
-  idxcg_count = 0;
-  for (int il = 0; il < nqlist; il++) {
-    int l = qlist[il];
-    for (int m1 = 0; m1 < 2 * l + 1; m1++) {
-      aa2 = m1 - l;
-      for (int m2 = MAX(0, l - m1); m2 < MIN(2 * l + 1, 3 * l - m1 + 1); m2++) {
-        bb2 = m2 - l;
-        m = aa2 + bb2 + l;
-
-        // clang-format off
-
-        sum = 0.0;
-        for (int z = MAX(0, MAX(-aa2, bb2));
-             z <= MIN(l, MIN(l - aa2, l + bb2)); z++) {
-          ifac = z % 2 ? -1 : 1;
-          sum += ifac /
-            (factorial(z) *
-             factorial(l - z) *
-             factorial(l - aa2 - z) *
-             factorial(l + bb2 - z) *
-             factorial(aa2 + z) *
-             factorial(-bb2 + z));
-        }
+/* ---------------------------------------------------------------------- */
 
-        cc2 = m - l;
-        sfaccg = sqrt(factorial(l + aa2) *
-                      factorial(l - aa2) *
-                      factorial(l + bb2) *
-                      factorial(l - bb2) *
-                      factorial(l + cc2) *
-                      factorial(l - cc2) *
-                      (2*l + 1));
-        // clang-format on
-
-        sfac1 = factorial(3 * l + 1);
-        sfac2 = factorial(l);
-        dcg = sqrt(sfac2 * sfac2 * sfac2 / sfac1);
-
-        cglist[idxcg_count] = sum * dcg * sfaccg;
-        idxcg_count++;
-      }
+double ComputeOrientOrderAtom::triangle_coeff(const int a, const int b, const int c) {
+  return factorial(a+b-c)*factorial(a-b+c)*factorial(-a+b+c) / factorial(a+b+c+1);
+}
+
+/* ---------------------------------------------------------------------- */
+
+double ComputeOrientOrderAtom::w3j(const int lmax, const int j1, const int j2, const int j3) {
+  const int a = lmax, b = lmax, c = lmax;
+  const int alpha = j1, beta = j2, gamma = j3;
+  struct {
+    double operator() (const int a, const int b, const int c,
+                       const int alpha, const int beta,const int gamma,
+                       const int t) {
+      return factorial(t)*factorial(c-b+t+alpha)*factorial(c-a+t-beta) * factorial(a+b-c-t)*factorial(a-t-alpha)*factorial(b-t+beta);
     }
+  } x;
+  const double
+    sgn = 1 - 2*((a-b-gamma)&1),
+    g = sqrt(triangle_coeff(lmax,lmax,lmax)) * sqrt(factorial(a+alpha)*factorial(a-alpha)*
```

Comment:
@akohlmey this new code isn't formatted--i.e. changing`(a+alpha)*factorial(a-alpha)` to `(a + alpha) * factorial(a - alpha)`

Reply:
`clang-format -i <filename1> <filename2> ...` should do the trick. I can do it for you, if needed.

Reply:
@akohlmey it would be great if you could clang-format this, thanks